batch/batch/utils.py

```diff
@@ -118,6 +122,29 @@ def adjust_cores_for_memory_request(cores_in_mcpu, memory_in_bytes, worker_type)
     return max(cores_in_mcpu, min_cores_mcpu)
 
 
+def total_worker_storage():
+    # local ssd is 375Gi
+    # reserve 25Gi for images
+    return 375 - 25
+
+
+def worker_storage_per_core_bytes(worker_cores):
+    return (total_worker_storage() * 1024**3) // worker_cores
+
+
+def storage_bytes_to_cores_mcpu(storage_in_bytes, worker_cores):
+    return math.ceil((storage_in_bytes / worker_storage_per_core_bytes(worker_cores)) * 1000)
+
+
+def cores_mcpu_to_storage_bytes(cores_in_mcpu, worker_cores):
+    return int((cores_in_mcpu / 1000) * worker_storage_per_core_bytes(worker_cores))
```

Comment:
I suspect these two functions will work fine, but floating point arithmetic gives me the heebie jeebies. I prefer:

```
(cores_in_mcpu * worker_storage_per_core_bytes(worker_cores)) // 1000
```
and
```
def round_up_division(items, group_size):
    return (items + group_size - 1) // group_size
# ...
1000 * round_up_division(storage_in_bytes, worker_storage_per_core_bytes(worker_cores))
```

I think I just find integer arithmetic less surprising and thus prefer it when possible. Moreover, the values in question here will divide evenly. Consider that the smallest CPU fraction we permit is `250mcpu`. 250 = 5 * 5 * 5 * 2. The worker storage per core bytes has at least 26 factors of two (30 from 1024**3 minus 4 from 16 cores). As such, our values for storage should always be divisible by `1000 = 5 * 5 * 5 * 2 * 2 * 2`.

Reply:
Where do you get that it's always divisible by 1000? i.e. where do the powers of 5 come in 2**26?

Reply:
This round_up_division function is dangerous!!! It rounds up 0.25 to 1.

Reply:
I suppose that's a bad name. It's something like `how_many_groups_do_i_need`.

`cores_in_mcpu`, unless I'm mistaken, is always a multiple of 250, ergo it always has three factors of five.

Ah, yes this would calculate the cores necessary, but we want the mcpu necessary for that much storage, ergo we should push the `1000 *` into the numerator:

```
round_up_division(1000 * storage_in_bytes, worker_storage_per_core_bytes(worker_cores))
```